### PR TITLE
Image content type validations

### PR DESCRIPTION
**Pull request by sudara**
Currently we aren't validating image content type for user avatars or playlists like we were before the Active Storage migration.

That's leading to issues like #1010 where `ActiveStorage::InvariableError` is raised.

I'm trying to reuse the Active Storage `attached` validator used for `Asset`, however the controller seems to hit AS logic before validations have a chance to run.

There was some chatter of a similar issue being fixed in 6.0, but `ActiveStorage::InvariableError` still seems to be hit before playlist validations are hit on this branch.

https://github.com/rails/rails/issues/32449#issuecomment-404938746

Any thoughts @Manfred?

### CODE REVIEW

File: app/models/playlist.rb [hunk 1]
```diff
@@ -34,6 +34,10 @@ class Playlist < ActiveRecord::Base
   # We have to define attachments last to make the Active Record callbacks
   # fire in the right order.
   has_one_attached :cover_image
+  validates :cover_image, attached: {
+    content_type: %w[image/png image/jpeg image/jpg image/gif],
+    byte_size: { less_than: 20.megabytes }
+  }, if: :cover_image_present?
```
Comment (sudara) [inline app/models/playlist.rb:40]: Cover images are optional, but even if i remove this if, the validation doesn't seem to run before it hits an AS error.

**Comment by Manfred** [comment]
I'm confused, this PR does not solve the issue? Or does it?

**Comment by sudara** [comment]
@Manfred I hoped this branch would solve the issue, but the problem seems deeper, so the PR is just a way to share code and context. Do you happen to remember if content type validation was problematic in the migration or if there was a technical reason that images aren't validated?

**Comment by Manfred** [comment]
The exception you are referring to is caused by the views attempting to do something with the invalid attachment. When the Playlist validation rejects the cover image it just sets an error on `:cover_image` and `save` returns false. Then the edit view attempts to render a cover image with whatever weird state `@playlist.cover_image` is in. So I guess the solution is to reset the original cover image somehow when there are errors on `:cover_image`.

**Pull request merged**